### ISSUE TITLE
Does the demo has shape only preprocessing layer in inference?

### ISSUE THREAD

**Issue by dragonfly90** [vicariousinc/science_rcn#8]
As mentioned in section 4.2.1 of the supplementary material: The preprocessing layer.

An even simpler option is to use a preprocessing stage that discards the appearance consistency information. Such preprocessing stage only needs to perform edge detection at multiple rotations (without considering for each rotation the three di↵erent orientations described in Fig. S2) and produces only a small performance degradation in practice. Any edge detection algorithm such as Gabor filtering can produce satisfactory results.

**Comment by cruyffturn** [comment]
Yes I believe so, I  think the Conditional Markov Field section and combining them with the Shape information exists in the reference implementation. 

In MNIST dataset it's not necessary I think in easier CAPTCHA datasets you can also get away without using the appearance information.
